tmf-transform.exs

```diff
@@ -0,0 +1,225 @@
+            # if the value has `\r\n` in it, split into a list of strings
+            value =
+              v
+              |> special_conversion_hack()
+              |> split_string_with_multiple_lines()
+
+            # Search for Section headers (from the headers) to create sub maps
+            k
+            |> has_section_marker?()
+            |> case do
+              true ->
+                # Need to split key into section/header
+                {true, add_section_to_artifact(acc, k, value)}
+              false ->
+                {true, acc |> Map.put(k, value)}
+            end
+          end)
+
+        artifact
+      end)
+
+    %{metadata: metadata, artifacts: artifacts}
+  end
+
+  defp has_section_marker?(str), do: String.contains?(str, " : ")
+
+  defp split_string_with_multiple_lines(str) when is_binary(str) do
+    str
+    |> String.split("\r\n")
+    |> case do
+      [any] -> any
+      any -> any
+    end
+  end
+
+  defp split_string_with_multiple_lines(other), do: other
+
+  # Handle unexpected formatting in a given XLSX version of the reference model
+  defp special_conversion_hack("2.2000000000000002"), do: "2.2"
+  defp special_conversion_hack("2.0099999999999998"), do: "2.01"
+  defp special_conversion_hack("10.050000000000001"), do: "10.05"
+  defp special_conversion_hack(any), do: any
+
+  defp add_section_to_artifact(artifact, key, value) do
+    # split key into section/header
+    [section, header] = String.split(key, " : ", trim: true, parts: 2)
+
+    {_, new_artifact} =
+      artifact
+      |> Map.get_and_update(section, fn
+        current_value when is_nil(current_value) ->
+          {current_value, Map.new([{header, value}])}
+
+        current_value ->
+          {current_value, current_value |> Map.put(header, value)}
+      end)
+
+    new_artifact
+  end
+end
+
+defmodule TmfReferenceModel.Encoder.Json do
+  @moduledoc """
+  For the output of the Transfomer module, convert the map
+  into a JSON representation.
+  """
+
+  @doc """
+  Given an Elixir Map, with `:metadata` and `:artifacts` keys,
+  convert it into it's JSON representation.
+  """
+  def encode(%{metadata: metadata, artifacts: artifacts}) do
+    %{
+      :_metadata => metadata,
+      artifacts: %{
+        :_count => artifacts |> Enum.count(),
+        items: artifacts
+      }
+    }
+    |> Jason.encode!(pretty: true)
+  end
+
+  def encode(_), do: raise ArgumentError, "Must supply a map with metadata and artifacts keys"
+end
+
+defmodule TmfReferenceModel.Main do
+  @moduledoc """
+  Main entry point to the script
+  """
+
+  alias TmfReferenceModel.Loader
+  alias TmfReferenceModel.Transformer
+  alias TmfReferenceModel.Encoder.Json
+
+  def main() do
+    input_file = "Version-3.2.1-TMF-Reference-Model-v01-Mar-2021.xlsx"
+    output_file = input_file |> String.replace("xlsx", "json")
+
+    input_file
+    |> Loader.load()
+    |> Transformer.transform()
+    |> Json.encode()
+    |> then(&File.write(output_file, &1))
```

Comment:
my initial thought is... "I love pipelines! They are core to my Elixir style!". 

I'll spend time reading that discussion and thinking about what's going on here, and how it can be more robust. Thank you for sharing.